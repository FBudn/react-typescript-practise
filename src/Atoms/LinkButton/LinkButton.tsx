/* eslint-disable react/button-has-type */
import React from "react";

export interface LinkButtonProps {
  children: React.ReactNode;
  onClickLink: any;
}

const LinkButton: React.FC<LinkButtonProps> = ({ children, onClickLink }) => {
  return <button onClick={onClickLink}>{children}</button>;
};

export default LinkButton;
