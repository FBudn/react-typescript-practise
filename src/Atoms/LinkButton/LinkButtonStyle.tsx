import styled from "styled-components";

export const LinkButtonStyled = styled.button`
  display: flex;
  border: none;
  justify-content: center;
  align-items: end;
  font-size: 1rem;
  font-weight: 500;
  color: rgb(82 82 91);
  cursor: pointer;
  text-decoration: underline;
`;
