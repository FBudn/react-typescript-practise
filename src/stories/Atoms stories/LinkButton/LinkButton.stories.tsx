import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import LinkButton from "../../../Atoms/LinkButton/LinkButton";

export default {
  title: "Atoms/LinkButton",
  component: LinkButton,
  argTypes: {
    onClickLink: { action: "Im switching page" },
  },
} as ComponentMeta<typeof LinkButton>;

const Template: ComponentStory<typeof LinkButton> = (args) => (
  <LinkButton {...args} />
);

export const BasicLOGIN = Template.bind({});
BasicLOGIN.args = {
  children: "LOGIN",
};

export const BasicSIGNUP = Template.bind({});
BasicSIGNUP.args = {
  children: "SIGN UP",
};
