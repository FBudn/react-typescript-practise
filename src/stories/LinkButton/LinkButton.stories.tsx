/* eslint-disable @typescript-eslint/consistent-type-assertions */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import { Meta, Story } from "@storybook/react";
import LinkButton, { LinkButtonProps } from "../../Atoms/LinkButton/LinkButton";

export default {
  title: "Atoms/LinkButton",
  component: LinkButton,
  argTypes: {
    onClickLink: { action: "Im switching page" },
  },
} as Meta;

const Template: Story<LinkButtonProps> = (args) => (
  <LinkButton {...args}>{args.children}</LinkButton>
);

export const Basic = Template.bind({});
Basic.args = {
  children: "Link which as a default has global text styles",
};
