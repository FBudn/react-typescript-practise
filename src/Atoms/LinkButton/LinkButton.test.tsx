import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import TestRenderer from "react-test-renderer";
import LinkButton, { LinkButtonProps } from "./LinkButton";
import "jest-styled-components";

const mockonClickLink = jest.fn();

const props: LinkButtonProps = {
  testId: `test-linkButton-id`,
  children: `LinkButton Test`,
  onClickLink: mockonClickLink,
};

test(`Should renred LinkButton component`, () => {
  render(<LinkButton {...props} />);
  const LinkButtonElement = screen.getByTestId(`test-linkButton-id`);
  expect(LinkButtonElement).toBeInTheDocument();
});

test(`Schould match snapshot`, () => {
  const LinkButtonElement = TestRenderer.create(
    <LinkButton {...props} />,
  ).toJSON();
  expect(LinkButtonElement).toMatchSnapshot();
});

test(`Should render and have styles:`, () => {
  render(<LinkButton {...props} />);
  const LinkButtonElement = screen.getByTestId(`test-linkButton-id`);
  expect(LinkButtonElement).toBeInTheDocument();

  expect(LinkButtonElement).toHaveStyleRule("display", "flex");
  expect(LinkButtonElement).toHaveStyleRule("border", "none");
  expect(LinkButtonElement).toHaveStyleRule("justify-content", "center");
  expect(LinkButtonElement).toHaveStyleRule("align-items", "end");
  expect(LinkButtonElement).toHaveStyleRule("justify-content", "center");
  expect(LinkButtonElement).toHaveStyleRule("align-items", "end");
  expect(LinkButtonElement).toHaveStyleRule("cursor", "pointer");
  expect(LinkButtonElement).toHaveStyleRule("text-decoration", "underline");
});

test(`Should handleClick`, () => {
  render(<LinkButton {...props} />);
  const LinkButtonElement = screen.getByTestId(`test-linkButton-id`);
  fireEvent.click(LinkButtonElement);
  expect(mockonClickLink).toBeCalledTimes(1);
  mockonClickLink(`LinkButton Test 1`);
  expect(mockonClickLink).toBeCalledWith(`LinkButton Test 1`);
});
