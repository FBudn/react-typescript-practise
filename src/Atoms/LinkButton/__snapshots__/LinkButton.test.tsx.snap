// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: none;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  font-size: 1rem;
  font-weight: 500;
  color: rgb(82 82 91);
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<button
  className="c0"
  data-testid="test-linkButton-id"
  onClick={[MockFunction]}
>
  LinkButton Test
</button>
`;
